solc/CommandLineInterface.cpp

```diff
@@ -845,52 +845,56 @@ void CommandLineInterface::handleCombinedJSON()
 	output[g_strVersion] = frontend::VersionString;
 	vector<string> contracts = m_compiler->contractNames();
 
+	// NOTE: The state checks here are more strict that in Standard JSON. There we allow
+	// requesting certain outputs even if compilation fails as long as analysis went ok.
+	bool compilationSuccess = m_compiler->state() >= CompilerStack::State::CompilationSuccessful;
```

Comment:
This is the same as `m_compiler->compilationSuccessful()` - so I'm just curious as to why you refactored it?